task2/src/components/AppErrorBoundary/AppErrorBoundary.jsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+class AppErrorBoundary extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { hasError: false };
+    }
+
+    static getDerivedStateFromError(error) {       
+        return { hasError: true };
+    }
+
+    componentDidCatch(error, errorInfo) {        
+        console.log(error, errorInfo);
+    }
+
+    render() {
+        if (this.state.hasError) {           
+            return <h1>Ohhh, no... Something went wrong.</h1>;
```

Comment:
u can extend your ErrorBoundary with one more not required prop e.g. `localBoundary` and use it everywhere if need

Reply:
localBoundary ?? <h1>Ohhh, no... Something went wrong.</h1>;